src/Uno.UI/UI/Xaml/Internal/VisualTree.cs

```diff
@@ -922,6 +922,24 @@ private static void VisualTreeNotFoundWarning()
 			}
 		}
 
+		internal event SizeChangedEventHandler SizeChanged
+		{
+			add
+			{
+				if (RootElement is FrameworkElement fe)
```

Comment:
This seems odd 🤔 - the screen size should directly affect `XamlRoot`, so size changes should not have to depend on the root element